hpx/runtime/parcelset/parcelport.hpp

```diff
@@ -370,11 +350,10 @@ namespace hpx { namespace parcelset
         /// mutex for all of the member data
         mutable lcos::local::spinlock mtx_;
 
-        /// The handler for all incoming requests.
-        server::parcelport_queue parcels_;
+        hpx::applier::applier *applier_;
 
         /// The cache for pending parcels
-        typedef std::pair<std::vector<parcel>, std::vector<write_handler_type> >
+        typedef std::pair<std::deque<parcel>, std::deque<write_handler_type> >
```

Comment:
Hmmm, this does not sound like a correct solution. Serializing every parcel separately limits the optimization capabilities we can employ. Couldn't we just force only those parcels which actually carry a future to be serialized on their own? Since we have to do two passes over the data (once for determining the buffer sizes and once for the actual serialization), this information could be retrieved without additional overhead before the serialization has begun.